### PR TITLE
Complete Australia Post driver

### PR DESCRIPTION
**Pull request by bradietilley**
Complete Australia Post driver

...

- [x] I have read the **[CONTRIBUTING](https://github.com/parceltrap/driver-auspost/blob/main/.github/CONTRIBUTING.md)** document.

### CODE REVIEW

File: src/AusPost.php [hunk 1]
```diff
@@ -4,65 +4,272 @@
...
+                    previous: $exception,
+                );
+            }
+
+            if (in_array($exception->getCode(), [401, 403])) {
+                throw new ApiAuthenticationFailedException(
+                    driver: $this,
+                    previous: $exception,
+                );
+            }
```
Comment (owenvoke) [inline src/AusPost.php:62]: Probably want to re-throw the exception here if it's not one of the other types. 👍🏻 That'll sort out the PHPStan warning.

**Review by owenvoke — CHANGES_REQUESTED** [review]
This looks great, one comment. Plus, we can bump the minimum ParcelTrap version in Composer to `^1.2.0` so that it always has the required exceptions. 👍🏻 Thanks mate.

File: src/AusPost.php (hunk 1 above)
Comment (bradietilley) [reply src/AusPost.php:62]: Good catch. Bumped up the min version to `^1.2`, added a re-throw, and added a test to ensure the rethrows still work :+1:

**Review by owenvoke — APPROVED** [review]
❤️

**Pull request merged**